crates/viewer/re_viewer_context/src/store_hub.rs

```diff
@@ -14,7 +14,54 @@ use re_query::CachesStats;
 use re_smart_channel::SmartChannelSource;
 use re_types::components::Timestamp;
 
-use crate::{BlueprintUndoState, Caches, StoreBundle, StoreContext};
+use crate::{
+    BlueprintUndoState, Caches, StoreBundle, StoreContext, TableId, TableStore, TableStores,
+};
+
+#[derive(Clone)]
+pub enum Entry {
+    Recording {
+        store_id: StoreId,
+        // TODO(grtlr): Add `applicationId` here.
+    },
+    Table {
+        table_id: TableId,
+    },
+}
+
+impl From<StoreId> for Entry {
+    fn from(store_id: StoreId) -> Self {
+        Self::Recording { store_id }
+    }
+}
+
+impl From<TableId> for Entry {
+    fn from(table_id: TableId) -> Self {
+        Self::Table { table_id }
+    }
+}
+
+impl Entry {
+    pub fn recording_ref(&self) -> Option<&StoreId> {
+        match self {
+            Self::Recording { store_id } => Some(store_id),
+            Self::Table { .. } => None,
+        }
+    }
+
+    pub fn table_ref(&self) -> Option<&TableId> {
+        match self {
+            Self::Table { table_id } => Some(table_id),
+            Self::Recording { .. } => None,
+        }
+    }
+}
+
+pub struct StorageContext<'a> {
```

Comment:
nit: would be nice to separate that for discoverability just like ViewerContext and StoreContext are separate